src/LrtSquare.sol

```diff
@@ -27,18 +30,27 @@ contract LrtSquare is
     ERC20PermitUpgradeable,
     UUPSUpgradeable
 {
+    using BucketLimiter for BucketLimiter.Limit;
     using SafeERC20 for IERC20;
     using Math for uint256;
 
     struct TokenInfo {
         bool registered;
         bool whitelisted;
+        uint64 maxPercentageInVault;
     }
 
     mapping(address => TokenInfo) public tokenInfos;
     mapping(address => bool) public depositor;
     address[] public tokens;
     address public priceProvider;
+    BucketLimiter.Limit private rateLimit;
+    address public rebalancer;
+    mapping(address swapOutputTokens => bool isWhitelisted) public isValidRebalanceOutputToken;
+    uint256 public maxAcceptableSlippageForRebalancing; // in 18 decimals
+    address public swapper; 
```

Comment:
could you add some comments on it? what is "swapper"?

Reply:
Sure